libsolidity/parsing/Token.cpp

```diff
@@ -50,6 +50,67 @@ namespace dev
 namespace solidity
 {
 
+void ElementaryTypeNameToken::parseDetails(Token::Value _baseType, unsigned const& _first, unsigned const& _second)
+{
+	solAssert(Token::isElementaryTypeName(_baseType), "");
+	string baseType = Token::toString(_baseType);
+	if (_first == 0 && _second == 0)
+	{
+		m_name = baseType;
+		m_firstNumber = _first;
+		m_secondNumber = _second;
+	}
+
+	if (baseType == "bytesM")
+	{
+		solAssert(_second == 0, "There should not be a second size argument to type bytesM.");
+		for (unsigned m = 1; m <= 32; m++)
+			if (m == _first)
+			{
+				m_name = baseType.substr(0, baseType.size()-1) + to_string(_first);
+				m_firstNumber = _first;
+				m_secondNumber = _second;
+			}
+	}
+	else if (baseType == "uintM" || baseType == "intM")
```

Comment:
Please do not compare the strings, this is very expensive and we are at the innermost part of the compiler here. Replacing these by something like `_baseType == Token::Value::UIntM` is semantically identical but much faster.